src/Form/FormControl.js

```diff
@@ -49,6 +50,19 @@ class FormControl extends Component {
     };
   }
 
+  componentWillMount() {
+    Children.forEach(this.props.children, child => {
+      if (
+        child &&
+        child.type &&
+        (child.type.name === 'Input' || (child.type.Naked && child.type.Naked.name === 'Input')) &&
```

Comment:
I'm not sure we want to rely on `Naked` here. What about using `muiName` instead? You can find other examples on the code base.